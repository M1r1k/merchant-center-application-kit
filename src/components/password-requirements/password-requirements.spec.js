import React from 'react';
import { shallow } from 'enzyme';
import PasswordRequirements from './password-requirements';
import styles from './password-requirements.mod.css';

const createTestProps = props => ({
  isExpanded: true,
  shouldShowUnfulfilledAsError: true,
  ...props,
});

const Child = () => <div />;

describe('rendering', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = createTestProps();
    wrapper = shallow(
      <PasswordRequirements {...props}>
        <Child />
      </PasswordRequirements>
    );
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  describe('when `shouldShowUnfulfilledAsError` is `true`', () => {
    beforeEach(() => {
      props = createTestProps({ shouldShowUnfulfilledAsError: true });
      wrapper = shallow(
        <PasswordRequirements {...props}>
          <Child />
        </PasswordRequirements>
      );
    });
    it('should pass on `shouldShowUnfulfilledAsError` to children', () => {
      expect(wrapper.find(Child)).toHaveProp(
        'shouldShowUnfulfilledAsError',
        true
      );
    });
  });
  describe('when `shouldShowUnfulfilledAsError` is `false`', () => {
    beforeEach(() => {
      props = createTestProps({ shouldShowUnfulfilledAsError: false });
      wrapper = shallow(
        <PasswordRequirements {...props}>
          <Child />
        </PasswordRequirements>
      );
    });
    it('should pass on `shouldShowUnfulfilledAsError` to children', () => {
      expect(wrapper.find(Child)).toHaveProp(
        'shouldShowUnfulfilledAsError',
        false
      );
    });
  });
  describe('when `isExpanded` is `true`', () => {
    beforeEach(() => {
      props = createTestProps({ isExpanded: true });
      wrapper = shallow(
        <PasswordRequirements {...props}>
          <Child />
        </PasswordRequirements>
      );
    });
    it('should pass on `shouldShowUnfulfilledAsError` to children', () => {
      expect(
        wrapper.find(`.${styles['password-requirements']}`)
      ).toHaveClassName(styles.expanded);
    });
  });
  describe('when `isExpanded` is `false`', () => {
    beforeEach(() => {
      props = createTestProps({ isExpanded: false });
      wrapper = shallow(
        <PasswordRequirements {...props}>
          <Child />
        </PasswordRequirements>
      );
    });
    it('should pass on `shouldShowUnfulfilledAsError` to children', () => {
      expect(
        wrapper.find(`.${styles['password-requirements']}`)
      ).not.toHaveClassName(styles.expanded);
    });
  });
});
