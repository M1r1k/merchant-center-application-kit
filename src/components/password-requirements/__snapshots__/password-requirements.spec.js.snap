// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should match snapshot 1`] = `
<div
  className="password-requirements expanded"
>
  <Child
    key=".0"
    shouldShowUnfulfilledAsError={true}
  />
</div>
`;
